src/cata_tiles.cpp

```diff
@@ -859,26 +860,29 @@ bool cata_tiles::draw_tile_at(tile_type *tile, int x, int y, int rota)
     return true;
 }
 
-bool cata_tiles::draw_lighting(int x, int y, LIGHTING l)
+bool cata_tiles::draw_lighting(int x, int y, lit_level l)
 {
     std::string light_name;
     switch(l) {
-        case HIDDEN:
+        case LL_BLANK:
             light_name = "lighting_hidden";
             break;
-        case LIGHT_NORMAL:
-            light_name = "lighting_lowlight_light";
-            break;
-        case LIGHT_DARK:
-            light_name = "lighting_lowlight_dark";
-            break;
-        case BOOMER_NORMAL:
-            light_name = "lighting_boomered_light";
+        case LL_LIT:
```

Comment:
You're only calling this function when `l` is one of `LL_DARK`, `LL_BRIGHT_ONLY` or `LL_BLANK`, why check for the other values?

The return value is currently ignored, you could change it as to indicate whether the area is actually visible (further tiles must be drawn) or not. Than you could move that decision from `draw` into this function, the code in `draw` would be:
```C++
// in the loop in cata_tiles::draw
if( !draw_lighting( ... ) ) {
    // draw the infrared critter, if any
    draw_critter...
    // and continue as the area is not visible at all.
    continue;
}
draw_terrain(...)
```

This has the advantage that you put all the light related decisions into one place. You could even go so far and move the call to `map::apparent_light_at` out of `draw` (it would not be used anymore) and into this function.